lib/spack/spack/build_systems/cmake.py

```diff
@@ -288,6 +291,15 @@ def std_args(pkg, generator=None):
                 [define("CMAKE_FIND_FRAMEWORK", "LAST"), define("CMAKE_FIND_APPBUNDLE", "LAST")]
             )
 
+        if getattr(pkg, "find_python_hints", False) and "python" in pkg.dependency_names():
```

Comment:
This is wrong. Don't use `pkg`, use `spec`. E.g. `if spec.dependencies("python")`.